addons/sale/models/payment_transaction.py

```diff
@@ -33,17 +33,22 @@ def _compute_sale_order_ids_nbr(self):
 
     def _set_pending(self, state_message=None):
         """ Override of payment to send the quotations automatically. """
+        premilinary_state = self.state
         super(PaymentTransaction, self)._set_pending(state_message=state_message)
 
-        for record in self:
-            sales_orders = record.sale_order_ids.filtered(lambda so: so.state in ['draft', 'sent'])
-            sales_orders.filtered(lambda so: so.state == 'draft').with_context(tracking_disable=True).write({'state': 'sent'})
-
-            if record.acquirer_id.provider == 'transfer':
-                for so in record.sale_order_ids:
-                    so.reference = record._compute_sale_order_reference(so)
-            # send order confirmation mail
-            sales_orders._send_order_confirmation_mail()
+        # When using webhooks, this function can be called two times (when the user comes back and
+        # when the webhooks notify the db). To avoid duplicates, check if the transaction is not
+        # already pending
+        if premilinary_state != 'pending' and  self.state == 'pending':
```

Comment:
Seems like a fix to be done in stable, did you try reproducing it in stable version (let's say 15+) ?

Reply:
I think that this is specifically linked to refunds but I'll check it in stable version to be sure

Reply:
Currently, I don't see any cases in Stable where `_set_pending` could be triggered two times. But since we've added webhooks in a lot of other tasks and acquirers in master, I think that it may be a good idea to change that in a separate commit because it's not linked directly to Stripe